pilot/pkg/serviceregistry/kube/controller/ambient/ambientindex.go

```diff
@@ -171,11 +171,19 @@ func New(options Options) Index {
 		ObjectFilter: kubetypes.ComposeFilters(options.Client.ObjectFilter(), a.inRevision),
 	}
 	opts := krt.NewOptionsBuilder(a.stop, "ambient", options.Debugger)
```

Comment:
in a follow up, it would be  nice not to need the ClusterKRTMetadata setup all over the place. 

We talked about it before and the worry was accidentally passing the local cluster ID to remotes IIRC.

Could we do

```go
func (k OptionsBuilder) WithMetadata(metadata Metadata) OptionsBuilder {
	if k.metadata != nil {
		k.metadata = maps.MergeCopy(k.metadata, metadata)
	} else {
		k.metadata = metadata
	}
	return k
}
```

and then have clearly distinct opts

```go
	opts := krt.NewOptionsBuilder(a.stop, "ambient", options.Debugger)
	localOpts := opts.WithMetadata(krt.Metadata{
		multicluster.ClusterKRTMetadataKey: options.ClusterID,
	})
```

Reply:
Love that idea; created #56719